pkg/virt-manifest/rest/manifest.go

```diff
@@ -0,0 +1,69 @@
+package rest
+
+import (
+	"github.com/emicklei/go-restful"
+	"golang.org/x/net/context"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/middleware"
+	"kubevirt.io/kubevirt/pkg/rest"
+	"kubevirt.io/kubevirt/pkg/rest/endpoints"
+	"kubevirt.io/kubevirt/pkg/virt-handler/virtwrap"
+	"kubevirt.io/kubevirt/pkg/virt-manifest"
+)
+
+type manifest struct {
+	libvirtUri    string
+	libvirtUser   string
+	libvirtPasswd string
+}
+
+func ManifestService(libvirtUri string, libvirtUser string, libvirtPasswd string) (*restful.WebService, error) {
+	ws := new(restful.WebService)
+
+	m := manifest{
+		libvirtUri:    libvirtUri,
+		libvirtUser:   libvirtUser,
+		libvirtPasswd: libvirtPasswd,
+	}
+
+	ctx := context.Background()
+	target := endpoints.MakeGoRestfulWrapper(endpoints.NewHandlerBuilder().Post(&v1.VM{}).Endpoint(m.mapManifest).Build(ctx))
+
+	ws.Path("/").
+		Consumes(restful.MIME_JSON, rest.MIME_YAML).
+		Produces(restful.MIME_JSON, rest.MIME_YAML)
+	ws.Route(ws.GET("/api/v1/status").To(alive))
+	ws.Route(ws.POST("/apis/" + v1.GroupVersion.String() + "/namespaces/{namespace}/manifest").To(target))
```

Comment:
So a little detail I forgot until I got to implementing it: In order to map persistent volume claims, we need a namespace. I'm going to re-add a namespace to the mapManifest path. If we want to support a non-namespaced URI for this endpoint, it would either need to assume the default namespace or it would not be able to map persistent volumes.